### PR TITLE
Saptune services status icons

### PR DESCRIPTION
**Pull request by EMaksy**
# Description

Adding Saptune Service Status  icons

## Demo
image

## How was this tested?
Added automated test

### CODE REVIEW

File: assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx [hunk 1]
```diff
@@ -14,42 +14,43 @@ import {
...
+    name: 'tuned',
+  };
+
+  const services = [
+    customSaptuneService,
+    customSapconfService,
+    customTunedService,
+  ];
 
+  const solution = saptuneSolutionFactory.build({ partial: false });
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx:41]: Why did you move these constants out?
I think they belong to the previous test.

For the 2nd test, you could use factory values:
```
const { configured_version: configuredVersion, enabled_solution: enabledSolution, applied_solution: appliedSolution } =
      saptuneStatusFactory.build();
```

I don't see having global constants as good idea here

File: assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx [hunk 2]
```diff
@@ -102,4 +105,35 @@ describe('SaptuneDetails', () => {
       staging.solutions_ids
     );
   });
+
+  it('should render all the icons properly', () => {
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx:109]: I see this test super weak. It should check all the combinations.
I would extract the service logic to `SaptuneService.jsx` component and test them all with a `it.each` kind of test.

**Review by arbulu89 — CHANGES_REQUESTED** [review]
I think we need to adjust the tests

File: assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx (hunk 1 above)
Comment (EMaksy) [reply assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx:41]: fair point i wanted to reuse the same data for the tests, but actually this was the moment where i should have just created a new component :D

File: assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx (hunk 2 above)
Comment (EMaksy) [reply assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx:109]: Good idea, it was challenging to test it, i will move it :D

File: assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ describe('SaptuneDetails', () => {
     };
 
     const customSapconfService = {
-      active: 'inactive',
+      active: 'active',
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx:25]: why changes in this file?

File: assets/js/components/SaptuneDetails/SaptuneServiceStatus.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+  return (
+    <div className="flex">
+      {icon}
+      <span className="ml-1">{text}</span>
+    </div>
+  );
+};
+
+function SaptuneServiceStatus({ serviceName, services }) {
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneServiceStatus.jsx:44]: I think this should be `{ serviceName, enabled, active}` and the next piece of code should remain in the `SaptuneDetails` file:
```
const currentService = find(services, { name: serviceName });
const { enabled, active } = currentService;
```
Passing the whole list of services looks really a smell (just seeing how strange the tests become)

File: assets/js/components/SaptuneDetails/SaptuneServiceStatus.test.jsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        name: service,
+      };
+
+      const services = [customService];
+
+      renderWithRouter(
+        <SaptuneServiceStatus services={services} serviceName={service} />
+      );
+
+      const serviceStatusIcons = screen.getAllByTestId('eos-svg-component');
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneServiceStatus.test.jsx:98]: `getByTestId`, we should only have one icon

File: assets/js/components/SaptuneDetails/SaptuneServiceStatus.test.jsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+      active: 'inactive',
+      icon: 'fill-jungle-green-500',
+    },
+  ])(
+    'should render the service and icon properly for service $service',
+    ({ service, enabled, active, icon }) => {
+      const customService = {
+        active,
+        enabled,
+        name: service,
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneServiceStatus.test.jsx:89]: You can name `name` above if this is the term we should in the factory

File: assets/js/components/SaptuneDetails/SaptuneDetails.jsx [hunk 7]
```diff
@@ -1,29 +1,15 @@
...
 
 import BackButton from '@components/BackButton';
 import ListView from '@components/ListView';
-
 import PageHeader from '@components/PageHeader';
 
 import SaptuneTuningState from './SaptuneTuningState';
 import SaptuneVersion from './SaptuneVersion';
-
-const renderService = (serviceName, services) => {
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneDetails.jsx:14]: I think we could keep this as:
```
const renderService = (serviceName, services) => {
  const { enabled, active } = find(services, { name: serviceName });
  return <SaptuneServiceStatus serviceName={serviceName} enabled={enabled} active={active} />;
}
```

**Review by arbulu89 — CHANGES_REQUESTED** [review]
Some suggestions @EMaksy

File: assets/js/components/SaptuneDetails/SaptuneDetails.jsx (hunk 7 above)
Comment (EMaksy) [reply assets/js/components/SaptuneDetails/SaptuneDetails.jsx:14]: Yeah sure i moved :D

File: assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx (hunk 3 above)
Comment (EMaksy) [reply assets/js/components/SaptuneDetails/SaptuneDetails.test.jsx:25]: it was a leftover from the previous test, so more  varied icons can be rendered.
I will revert it

File: assets/js/components/SaptuneDetails/SaptuneServiceStatus.jsx (hunk 4 above)
Comment (EMaksy) [reply assets/js/components/SaptuneDetails/SaptuneServiceStatus.jsx:44]: Yeah sure this is cleaner :D

File: assets/js/components/SaptuneDetails/SaptuneServiceStatus.jsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  },
+};
+
+function SaptuneServiceStatus({ service, enabled, active }) {
+  if (!enabled) {
+    return <span>-</span>;
+  }
+
+  const status = `${enabled}/${active}`;
+  const icon = get(servicesIcons, [service.name, status], null);
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneServiceStatus.jsx:32]: `service.name` ?, why don't we pass the `serviceName` directly here?

File: assets/js/components/SaptuneDetails/SaptuneServiceStatus.test.jsx [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+import '@testing-library/jest-dom';
+
+import { renderWithRouter } from '@lib/test-utils';
+
+import SaptuneServiceStatus from './SaptuneServiceStatus';
+
+describe('SaptuneServiceStatus', () => {
+  it.each([
+    {
+      service: { name: 'saptune', enabled: 'enabled', active: 'active' },
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneServiceStatus.test.jsx:12]: I don't think we need this `service` subtree.
Just `serviceName`, `enabled` and `active` should be alright.
Once you adjust the code to get the `serviceName` it will be ok

File: assets/js/components/SaptuneDetails/SaptuneDetails.stories.jsx [hunk 10]
```diff
@@ -119,3 +119,37 @@ export const StagingDisabled = {
...
+  },
+};
+export const SaptuneServiceStatusFailing = {
+  args: {
+    ...Default.args,
+
+    services: [
+      { name: 'saptune', enabled: 'disabled', active: 'inactive' },
+      { name: 'sapconf', enabled: 'disabled', active: 'active' },
+      { name: 'tuned', enabled: null, active: null },
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneDetails.stories.jsx:140]: Is `null/null` failing?

**Review by arbulu89 — CHANGES_REQUESTED** [review]
More things

File: assets/js/components/SaptuneDetails/SaptuneServiceStatus.jsx (hunk 8 above)
Comment (EMaksy) [reply assets/js/components/SaptuneDetails/SaptuneServiceStatus.jsx:32]: yeah much better!

File: assets/js/components/SaptuneDetails/SaptuneDetails.stories.jsx (hunk 10 above)
Comment (EMaksy) [reply assets/js/components/SaptuneDetails/SaptuneDetails.stories.jsx:140]: No there is no failing state for tuned, i thought better nothing then something in the failing case :D 

`  tuned: {
    'enabled/active': <HealthIcon health="warning" />,
    'enabled/inactive': <HealthIcon health="warning" />,
    'disabled/active': <HealthIcon health="warning" />,
    'disabled/inactive': <HealthIcon health="passing" />,
  },`
  
  Or what do you think? i also can delete it :D

Comment (arbulu89) [reply assets/js/components/SaptuneDetails/SaptuneDetails.stories.jsx:140]: Okey, it is fine them

File: assets/js/components/SaptuneDetails/SaptuneServiceStatus.test.jsx [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+      icon: 'fill-yellow-500',
+    },
+    {
+      serviceName: 'tuned',
+      enabled: 'disabled',
+      active: 'inactive',
+      icon: 'fill-jungle-green-500',
+    },
+  ])(
+    'should render the service and icon properly for service $service.serviceName',
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneServiceStatus.test.jsx:84]: This looks wrong. It should be `$serviceName`

File: assets/js/components/SaptuneDetails/SaptuneServiceStatus.test.jsx [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+    ({ serviceName, enabled, active, icon }) => {
+      renderWithRouter(
+        <SaptuneServiceStatus
+          serviceName={serviceName}
+          enabled={enabled}
+          active={active}
+        />
+      );
+
+      const serviceStatusIcons = screen.getByTestId('eos-svg-component');
```
Comment (arbulu89) [inline assets/js/components/SaptuneDetails/SaptuneServiceStatus.test.jsx:94]: `serviceStatusIcon` in singular

**Review by arbulu89 — APPROVED** [review]
Okey!
All good now.
Some minimal changes, but you can merge without other review.
